src/main/java/io/stargate/sgv2/jsonapi/service/embedding/configuration/PropertyBasedEmbeddingServiceConfig.java

```diff
@@ -56,6 +59,17 @@ public interface VertexAiConfig {
     String apiKey();
   }
 
+  public interface CohereConfig {
```

Comment:
Thought of making it as one, but wasn't sure because some embedding services will have more that the 3 provided details. Eg: vertexAi may get location and project id, Amazon Titan may get additional embedding size property(supports dynamically passing it). Should we handle this when handling createCollection changes which to be discussed today? apiKey having "" will remove it.

Reply:
Oh sorry, what I meant was a common base interface, not common to everything.
I 100% agree some will need more than others.

But there's no real hurry, we can always refactor once we have better idea of what all is needed.